### ISSUE TITLE
Problems building pFUnit_demos

### ISSUE THREAD

**Issue by ZedThree** [Goddard-Fortran-Ecosystem/pFUnit#102]
I'm currently having a couple of issues with building the pFUnit demos:

1. The `FARGPARSE_TOP_DIR` variable in `PFUNIT.mk` is empty, so building the demos don't work:
    ```
    /home/Tools/pFUnit_demos/Trivial: ./build_with_make_and_run.x 
    /home/Tools/pFUnit-4.0/build/install/PFUNIT-4.0/include/PFUNIT.mk:10: 
    /include/FARGPARSE.mk: No such file or directory
    make: *** No rule to make target '/include/FARGPARSE.mk'.  Stop.
    ./build_with_make_and_run.x: line 9: ./my_tests: No such file or directory
    ```
    Fargparse does get installed though, in `/path/to/install/FARGPARSE-0.9`

    Transitively going through `PFUNIT.mk`, `FARGPARSE.mk`, and `GFTL_SHARED.mk` to set their dependency paths correctly seems to fix things. But from the looks of things, these should be set at configure time?

2. Perhaps just my opinion, but I think you're installing pFUnit (and the dependencies) in the wrong place: if I run `cmake .. -DCMAKE_INSTALL_PREFIX=/path/to/install`, I would expect everything to be under `/path/to/install`, but it's actually under `/path/to/install/PFUNIT-4.0`. This is a bit surprising! I have to set `PFUNIT_DIR` to `/path/to/install/PFUNIT-4.0`, rather than the path I told CMake. Maybe the modules should go under `/path/to/install/includes/pfunit` given the number of them, but I think "principle of least surprise" would be good here -- just install stuff in the conventional paths.

**Comment by tclune** [comment]
1) Yes - I discovered this issue last night.   I have a fix, but have not had a chance to commit.   Was not noticed initially because I had been building the external dependencies, er, externally and in that case the TOP_DIR vars were correctly filled.

I will definitely fix before tomorrow, as I'm giving a webinar and want things to work.   Just could not fix last night because I had to finish the slides which were due today.

2. Hmm.  It works for me when I set PFUNIT_DIR to the parent directory.   Not saying it does not work the other way as well.     The problem is that I need to put the external dependencies somewhere if they are built as subdirs rather than as external packages.  (For my day job I need the latter because the external packages are used in other layers as well.)   My current choice is more consistent with that.    Also, I am not sure it is a good idea to put, say GFTL-1.1  under the pfunit subdir.     

Perhaps most importantly, this naming strategy works with CMAKE_PREFIX_PATH, which allows me to specify just one path for a build that involves many dependencies.

I do not claim expertise here, but I did need to do a lot of trial and error to get this to work with different options for the external dependencies.   I just dropped the ball for the demos.
